lab/futurize.py

```diff
@@ -8,11 +8,47 @@
 import pytest
 
 
+def monkey_patch_support_for_python2():
+    from future import standard_library
+    standard_library.install_aliases()
+
+    try:
+        # Monkey patch the backported ipaddress module with
+        # alternative versions that accept string addresses, in
+        # addition to unicode addresses, in python2 code.
+        #
+        # This keep compatibility simple. Slightly complicated by the
+        # fact that some of these classes inherit from each other.
+        import ipaddress
+
+        def python2_compat(class_, bases=()):
+            def wrapper(function):
```

Comment:
Why do you need this extra `wrapper` closure?
Change out the whole `function` thing for `class_.__init__` in your re-def of `__init__` no?

```python
def python2_compat(class_, bases=()):
     def __init__(self, address, *args, **kwargs):
         if isinstance(address, basestring):
             address = address.decode('utf-8')
         return class_.__init__(self, address, *args, **kwargs)

     return type(class_.__name__, (class_,) + bases, {'__init__': __init__)})
```